contracts/vending-minter/src/contract.rs

```diff
@@ -65,6 +65,17 @@ pub fn instantiate(
         .querier
         .query_wasm_smart(factory.clone(), &Sg2QueryMsg::Params {})?;
 
+    // Limit per address limit to 1% of num tokens
+    if (msg.init_msg.num_tokens > 100 && msg.init_msg.num_tokens < 5000)
+        && msg.init_msg.per_address_limit > (msg.init_msg.num_tokens / 100)
```

Comment:
Having trouble understanding this logic and hardcoded values. A comment could help.